cmd/autoscaler/main.go

```diff
@@ -58,6 +60,8 @@ const (
 var (
 	masterURL  = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
 	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+
+	endpointsInformer corev1informers.EndpointsInformer
```

Comment:
is there any reason why _this_ particilar informer is global?

Reply:
It is used in `uniScalerFactory`. I don't have a good way to pass it to `uniScalerFactory` without changing the interface. And I don't think we should change the interface.

Reply:
This is very brittle interface, prone to bugs though :beetle:

Reply:
Any suggestion?

Reply:
I wrap `uniScalerFactory` func into a func with `endpointsInformer` injected. PTAL.